arrow-fx/src/main/kotlin/arrow/fx/internal/LinkedMap.kt

```diff
@@ -0,0 +1,63 @@
+package arrow.fx.internal
+
+import arrow.core.Tuple2
+import arrow.core.getOption
+import arrow.core.toT
+
+/**
+ * A Map which tracks the insertion order of entries, so that entries may be
+ * traversed in the order they were inserted.
+ */
+internal class LinkedMap<K, V>(
```

Comment:
Can we use [`LinkedHashMap`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-linked-hash-map/index.html#kotlin.collections.LinkedHashMap) from `kotlin.collections` instead?

Reply:
Nope AFAIK :D Actually the current MVar has already that kind of list